packages/core/src/compiler/Style.ts

```diff
@@ -608,6 +687,126 @@ const couldMatchRels = (
   return true;
 };
 
+/**
+ * Helper fxn for @function isValidRelSubst.
+ *
+ * Determines if all the variables in @param sebind
+ * exist as valid keys in @param subst.
+ */
+const sebindExistsInSubst = (sebind: SEBind, subst: Subst): boolean => {
+  if (sebind.contents.tag !== "StyVar") {
+    throw new Error("expected sty var");
+  }
+  const validKeywords = Object.keys(subst);
+  return validKeywords.includes(sebind.contents.contents.value);
+};
+
+/**
+ * Helper fxn for @function isValidRelSubst.
+ *
+ * Determines if all the variables in @param expr
+ * exist as valid keys in @param subst.
+ */
+const selExprExistsInSubst = (expr: SelExpr, subst: Subst): boolean => {
+  if (expr.tag === "SEBind") {
+    return sebindExistsInSubst(expr, subst);
+  } else if (expr.tag === ("SEFunc" || "SEValCons" || "SEFuncOrValCons")) {
+    return expr.args.every((arg) => selExprExistsInSubst(arg, subst));
+  } else throw new Error("unknown tag");
+};
+
+/**
+ * Helper fxn for @function isValidRelSubst.
+ *
+ * Determines if all the variables in @param arg
+ * exist as valid keys in @param subst.
+ */
+const predArgExistsInSubst = (arg: PredArg, subst: Subst): boolean => {
+  if (arg.tag === "RelPred") {
+    return isValidRelSubst(subst, arg);
+  } else if (arg.tag === "SEBind") {
+    return sebindExistsInSubst(arg, subst);
+  } else throw new Error("unknown tag");
+};
+
+/**
+ * Determines if all the variables in @param rel exist as valid
+ * keys in @param subst.
+ *
+ * Ex, IsSubset(x,y) must have 'x' and 'y' exist as keys in @param subst
+ * for the relation to be valid.
+ */
+const isValidRelSubst = (subst: Subst, rel: RelationPattern): boolean => {
+  if (rel.tag === "RelPred") {
+    return rel.args.every((arg) => predArgExistsInSubst(arg, subst));
+  } else if (rel.tag === "RelBind") {
+    return false; // these don't have aliases
+  } else {
```

Comment:
There're no other possibilities. Why are we throwing here? TypeScript should complain about `boolean | undefined` doesn't match with `boolean` whenever you miss a case.

Reply:
Same comment goes to functions above this

Reply:
Since my cases are using `if/else if` clauses, typescript complains that I haven't checked for other `else` cases (even though there are none), and that the return type `boolean` should instead be `boolean | undefined`. The error throwing is mostly to make typescript happy about the typing of the overall function (I could also replace the second `else if` case with `else`, but perhaps the function is more clear this way with explicit cases?)

<img width="833" alt="Screen Shot 2021-07-27 at 2 54 59 PM" src="https://user-images.githubusercontent.com/59301744/127211644-7dbf6fba-ae2c-4442-8676-e6396296223f.png">